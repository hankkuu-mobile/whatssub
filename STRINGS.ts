import * as Localization from 'expo-localization';

import i18n from 'i18n-js';

const en = {
  SIGN_IN_WITH_GOOGLE: 'Sign in with Google',
  SIGN_IN_WITH_FACEBOOK: 'Sign in with Facebook',
  INTRO_TITLE_1: 'Regular payment',
  INTRO_TITLE_2: 'Subscription',
  INTRO_TITLE_3: 'Automatic transfer',
  INTRO_TITLE_4: 'Monthly fixed charge',
  INTRO_TITLE_5: 'Regular delivery',
  INTRO_MESSAGE: 'Do not forget\nBe smart',
  INTRO_WHATSSUB: 'Whats-Sub',
  NAVIGATE: 'Navigate',
  CHANGE_THEME: 'Change theme',
  PRODUCTION_CARD_SUBSCRIPTION_SWITCH_ON: 'UnSub',
  PRODUCTION_CARD_SUBSCRIPTION_SWITCH_OFF: 'Sub',
  AMOUNT_TO_BE_PAID: 'Amount to be paid',
  CATEGORY_CULTURE_LEISURE: 'Culture/Leisure',
  CATEGORY_EDUCATION: 'Education',
  CATEGORY_GAME: 'Game',
  EMPTY_WELCOME: 'WelCome',
  EMPTY_COMMENT1: 'Press + button',
  EMPTY_COMMENT2: 'Register the service in use',
  SETTING: 'Setting',
  SETTING_ACCOUNT: 'Account',
  SETTING_NOTIFICATION: 'Notification',
  SETTING_OTHERS: 'Others',
  SETTING_EMAIL: 'Email',
  SETTING_NOTIFICATION_BEFORE_PAYMENT: 'Pre-billing alarm',
  SETTING_NOTIFICATION_MARKETING_EMAIL: 'Marketing email notification',
  SETTING_NOTIFICATION_MARKETING_PUSH: 'Marketing push notification',
  SETTING_OTHERS_CONTACTUS: 'Contact us',
  SIGN_OUT: 'Sign Out',
};

const ko = {
  SIGN_IN_WITH_GOOGLE: '구글 계정으로 시작하기',
  SIGN_IN_WITH_FACEBOOK: '페이스북 계정으로 시작하기',
  INTRO_TITLE_1: '정기결제',
  INTRO_TITLE_2: '구독',
  INTRO_TITLE_3: '자동이체',
  INTRO_TITLE_4: '월 정액',
  INTRO_TITLE_5: '정기배송',
  INTRO_MESSAGE: '잊지말고\n똑똑하게',
  INTRO_WHATSSUB: '왓-썹',
  NAVIGATE: '이동하기',
  CHANGE_THEME: '테마 변경',
  PRODUCTION_CARD_SUBSCRIPTION_SWITCH_ON: '해지',
  PRODUCTION_CARD_SUBSCRIPTION_SWITCH_OFF: '구독',
  AMOUNT_TO_BE_PAID: '결제 예정 금액',
  CATEGORY_CULTURE_LEISURE: '문화/여가',
  CATEGORY_EDUCATION: '교육',
  CATEGORY_GAME: '게임',
  EMPTY_WELCOME: '반갑습니다',
  EMPTY_COMMENT1: '+ 버튼을 눌러',
  EMPTY_COMMENT2: '사용중인 서비스를 등록해 볼까요',
  SETTING: '설정',
  SETTING_ACCOUNT: '계정',
  SETTING_NOTIFICATION: '알림 설정',
  SETTING_OTHERS: '기타',
  SETTING_EMAIL: '이메일',
  SETTING_NOTIFICATION_BEFORE_PAYMENT: '결제 전 알람 설정',
  SETTING_NOTIFICATION_MARKETING_EMAIL: '마케팅 이메일 알림',
  SETTING_NOTIFICATION_MARKETING_PUSH: '마케팅 앱 푸쉬 알림',
  SETTING_OTHERS_CONTACTUS: '문의하기',
  SIGN_OUT: '로그아웃',
};

i18n.fallbacks = true;
i18n.translations = { en, ko };
i18n.locale = Localization.locale;

export const getString = (param: string, mapObj?: object) => {
  if (mapObj) {
    i18n.t(param, mapObj);
  }
  return i18n.t(param);
};
