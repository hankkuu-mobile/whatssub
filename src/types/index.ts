import { DefaultTheme } from 'styled-components';
/**
 * Fix error typing in `split` method in `apollo-link`
 * Related issue https://github.com/apollographql/apollo-client/issues/3090
 */
export interface Definition {
  kind: string;
  operation?: string;
};

export enum Gender { MALE, FEMALE };

export interface SocialInput {
  social: string;
  photo?: string;
  email?: string;
  password?: string;
  name?: string;
  nickname?: string;
  birthday?: Date;
  gender?: Gender;
  phone?: string;
}

export interface User {
  id: string;
  email: string;
  name: String;
  nickname: String;
  photo: String;
  birthday: Date;
  gender: Gender;
  social: String;
  verified: Boolean;
  notifications: [Notification]
  // customServices: [CustomService]
  // userProducts: [UserProduct]
  // reviews: [Review]
  created: Date
  updated: Date
  deleted: Date
}

export interface AuthPayload {
  token: string
  user: User
}

export enum ThemeType {
  LIGHT = 'LIGHT',
  DARK = 'DARK',
};

export interface ScreenProps {
  theme: DefaultTheme;
  changeTheme: () => void;
}
