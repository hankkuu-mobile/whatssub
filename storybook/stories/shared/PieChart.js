import { Text, TouchableOpacity } from 'react-native';

import PieChart from '../../../src/components/shared/PieChart';
import React from 'react';
import { getString } from '../../../STRINGS';
import { storiesOf } from '@storybook/react-native';
import styled from 'styled-components/native';

const Container = styled.SafeAreaView`
  flex: 1;
  align-items: center;
  justify-content: center;
`;
const ContainerDeco = (storyFn) => <Container>{storyFn()}</Container>;

const Story = () => {
  const [selectedMonth, setSelectedMonth] = React.useState('2019.06');

  return (
    <>
      <PieChart
        data={sampleDatas[selectedMonth]}
        style={{ width: 300, height: 300 }}
        defaultShowSlice={0}
        currentMonth={selectedMonth}
      />
      <TouchableOpacity
        onPress={() => {
          setSelectedMonth('2019.06');
        }}
      >
        <Text>{'2019.06'}</Text>
      </TouchableOpacity>
      <TouchableOpacity
        onPress={() => {
          setSelectedMonth('2019.07');
        }}
      >
        <Text>{'2019.07'}</Text>
      </TouchableOpacity>
    </>
  );
};

storiesOf('shared-PieChart', module)
  .addDecorator(ContainerDeco)
  .add('default', () => <Story />);

const sampleDatas = {
  2019.06: [
    {
      name: 'prod_1',
      category: getString('CATEGORY_CULTURE_LEISURE'),
      price: 100,
      currency: '',
      date: new Date(),
      image: '',
      currentMonthPaymentDate: new Date(),
    },
    {
      name: 'prod_2',
      category: getString('CATEGORY_CULTURE_LEISURE'),
      price: 100,
      currency: '',
      date: new Date(),
      image: '',
      currentMonthPaymentDate: new Date(),
    },
    {
      name: 'prod_3',
      category: getString('CATEGORY_CULTURE_LEISURE'),
      price: 100,
      currency: '',
      date: new Date(),
      image: '',
      currentMonthPaymentDate: new Date(),
    },
    {
      name: 'prod_4',
      category: getString('CATEGORY_EDUCATION'),
      price: 100,
      currency: '',
      date: new Date(),
      image: '',
      currentMonthPaymentDate: new Date(),
    },
    {
      name: 'prod_5',
      category: getString('CATEGORY_GAME'),
      price: 100,
      currency: '',
      date: new Date(),
      image: '',
      currentMonthPaymentDate: new Date(),
    },
  ],
  2019.07: [
    {
      name: 'prod_1',
      category: getString('CATEGORY_CULTURE_LEISURE'),
      price: 100,
      currency: '',
      date: new Date(),
      image: '',
      currentMonthPaymentDate: new Date(),
    },
    {
      name: 'prod_3',
      category: getString('CATEGORY_CULTURE_LEISURE'),
      price: 200,
      currency: '',
      date: new Date(),
      image: '',
      currentMonthPaymentDate: new Date(),
    },
    {
      name: 'prod_4',
      category: getString('CATEGORY_EDUCATION'),
      price: 100,
      currency: '',
      date: new Date(),
      image: '',
      currentMonthPaymentDate: new Date(),
    },
    {
      name: 'prod_5',
      category: getString('CATEGORY_GAME'),
      price: 100,
      currency: '',
      date: new Date(),
      image: '',
      currentMonthPaymentDate: new Date(),
    },
    {
      name: 'prod_6',
      category: getString('CATEGORY_GAME'),
      price: 200,
      currency: '',
      date: new Date(),
      image: '',
      currentMonthPaymentDate: new Date(),
    },
  ],
};
