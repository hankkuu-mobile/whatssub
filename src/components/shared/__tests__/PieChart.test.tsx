import * as React from 'react';

import { Text, TouchableOpacity } from 'react-native';

import PieChart from '../PieChart';
import { ThemeProvider } from 'styled-components/native';
import { ThemeType } from '../../../types';
import { createTheme } from '../../../theme';
import { getString } from '../../../../STRINGS';
import renderer from 'react-test-renderer';

const ComponentToTest = () => {
  const [selectedMonth, setSelectedMonth] = React.useState('2019.06');

  return (
    <ThemeProvider theme={createTheme(ThemeType.LIGHT)}>
      <>
        <PieChart
          data={sampleDatas[selectedMonth]}
          style={{ width: 300, height: 300 }}
          defaultShowSlice={0}
          currentMonth={selectedMonth}
        />
        <TouchableOpacity
          onPress={() => {
            setSelectedMonth('2019.06');
          }}
        >
          <Text>{'2019.06'}</Text>
        </TouchableOpacity>
        <TouchableOpacity
          onPress={() => {
            setSelectedMonth('2019.07');
          }}
        >
          <Text>{'2019.07'}</Text>
        </TouchableOpacity>
      </>
    </ThemeProvider>
  );
};

// wrap `ComponentToTest` to avoid Hooks bug
const component = () => <ComponentToTest />;

describe('[PieChart]', () => {
  it('renders without crashing', () => {
    const rendered = renderer.create(component()).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});

const sampleDatas = {
  2019.06: [
    {
      name: 'prod_1',
      category: getString('CATEGORY_CULTURE_LEISURE'),
      price: 100,
      currency: '',
      date: new Date(),
      image: '',
      currentMonthPaymentDate: new Date(),
    },
    {
      name: 'prod_2',
      category: getString('CATEGORY_CULTURE_LEISURE'),
      price: 100,
      currency: '',
      date: new Date(),
      image: '',
      currentMonthPaymentDate: new Date(),
    },
    {
      name: 'prod_3',
      category: getString('CATEGORY_CULTURE_LEISURE'),
      price: 100,
      currency: '',
      date: new Date(),
      image: '',
      currentMonthPaymentDate: new Date(),
    },
    {
      name: 'prod_4',
      category: getString('CATEGORY_EDUCATION'),
      price: 100,
      currency: '',
      date: new Date(),
      image: '',
      currentMonthPaymentDate: new Date(),
    },
    {
      name: 'prod_5',
      category: getString('CATEGORY_GAME'),
      price: 100,
      currency: '',
      date: new Date(),
      image: '',
      currentMonthPaymentDate: new Date(),
    },
  ],
  2019.07: [
    {
      name: 'prod_1',
      category: getString('CATEGORY_CULTURE_LEISURE'),
      price: 100,
      currency: '',
      date: new Date(),
      image: '',
      currentMonthPaymentDate: new Date(),
    },
    {
      name: 'prod_3',
      category: getString('CATEGORY_CULTURE_LEISURE'),
      price: 200,
      currency: '',
      date: new Date(),
      image: '',
      currentMonthPaymentDate: new Date(),
    },
    {
      name: 'prod_4',
      category: getString('CATEGORY_EDUCATION'),
      price: 100,
      currency: '',
      date: new Date(),
      image: '',
      currentMonthPaymentDate: new Date(),
    },
    {
      name: 'prod_5',
      category: getString('CATEGORY_GAME'),
      price: 100,
      currency: '',
      date: new Date(),
      image: '',
      currentMonthPaymentDate: new Date(),
    },
    {
      name: 'prod_6',
      category: getString('CATEGORY_GAME'),
      price: 200,
      currency: '',
      date: new Date(),
      image: '',
      currentMonthPaymentDate: new Date(),
    },
  ],
};
