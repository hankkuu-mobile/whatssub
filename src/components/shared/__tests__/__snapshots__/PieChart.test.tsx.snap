// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[PieChart] renders without crashing 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "justifyContent": "center",
        },
        Object {
          "height": 300,
          "width": 300,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "flexDirection": "column",
          },
          Object {
            "height": 50,
            "left": "75%",
            "position": "absolute",
            "top": 0,
            "width": 100,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "'rgb(62,126,255)'",
              "fontSize": 14,
            },
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#99a2aa",
              "fontSize": 14,
            },
          ]
        }
      >
        0원
      </Text>
    </View>
    <View
      style={
        Object {
          "height": 214,
          "width": 214,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "position": "absolute",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#323b43",
              "fontSize": 14,
            },
            Object {},
          ]
        }
      >
        2019.06
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#323b43",
              "fontSize": 14,
            },
            Object {
              "color": "#99a2aa",
              "fontSize": 14,
            },
          ]
        }
      >
        Amount to be paid
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#323b43",
              "fontSize": 14,
            },
            Object {
              "color": "#99a2aa",
              "fontSize": 24,
            },
          ]
        }
      >
        500원
      </Text>
    </View>
  </View>,
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text>
      2019.06
    </Text>
  </View>,
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text>
      2019.07
    </Text>
  </View>,
]
`;
